src/sentry/api/serializers/rest_framework/rule.py

```diff
@@ -168,31 +177,6 @@ def validate_conditions(self, conditions):
     def validate(self, attrs):
         return super().validate(validate_actions(attrs))
 
-    def save(self, rule):
```

Comment:
No code ran/referenced this ._.

Reply:
'o' is `.save()` run implicitly when people instantiate the serializer ? I couldn't find any explicit calls for `.save()`, and this method doesn't implement any of the `create()` or `update()`. So, we should be safe ?

Reply:
I am not sure -- I thought it was magically invoked by forms somehow -- could probably trace when this function was introduced and see how it was used in that PR and then from there figure out if that code path is still used

Reply:
:0) good thinking, looks like this (vintage) function used to be used in the rules endpoints, but now we use dataclasses (see: [ProjectRuleCreator](https://github.com/getsentry/sentry/blob/master/src/sentry/projects/project_rules/creator.py) & [ProjectRuleUpdater](https://github.com/getsentry/sentry/blob/master/src/sentry/projects/project_rules/updater.py#L13)) that do the same actions instead